src/test/operators/insert_test.cpp

```diff
@@ -44,14 +44,18 @@ TEST_F(OperatorsInsertTest, SelfInsert) {
   context->commit();
 
   // Check that row has been inserted.
-  EXPECT_EQ(t->get_chunk(ChunkID{0})->size(), 6u);
+  EXPECT_EQ(t->row_count(), 6u);
+  EXPECT_EQ(t->get_chunk(ChunkID{0})->size(), 3u);
+  EXPECT_EQ(t->get_chunk(ChunkID{1})->size(), 3u);
   EXPECT_EQ((*t->get_chunk(ChunkID{0})->get_segment(ColumnID{1}))[0], AllTypeVariant(12345));
   EXPECT_EQ((*t->get_chunk(ChunkID{0})->get_segment(ColumnID{0}))[0], AllTypeVariant(458.7f));
-  EXPECT_EQ((*t->get_chunk(ChunkID{0})->get_segment(ColumnID{1}))[3], AllTypeVariant(12345));
-  EXPECT_EQ((*t->get_chunk(ChunkID{0})->get_segment(ColumnID{0}))[3], AllTypeVariant(458.7f));
+  EXPECT_EQ((*t->get_chunk(ChunkID{1})->get_segment(ColumnID{1}))[0], AllTypeVariant(12345));
```

Comment:
This is because chunk 0 was previously not marked as immutable, correct?

Reply:
Correct. Before completion load_table marks the last chunk as immutable. Therefor, Insert has to add a new chunk.